### PR TITLE
MAGENTO-2255 - append child_id for simple products based on configurable

### PR DESCRIPTION
**Pull request by awesselburg**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php [hunk 1]
```diff
@@ -49,4 +50,21 @@ public function execute(Varien_Event_Observer $observer)
...
+    /**
+     * @param Mage_Wishlist_Model_Item $item
+     */
+    private function appendChildId($item)
+    {
+        $buyRequestItem = new Varien_Object($item->getBuyRequest()->getData());
+        $superAttribute = $buyRequestItem->getSuperAttribute();
+        if (is_array($superAttribute) && count($superAttribute)) {
+            $parentProduct = Mage::getModel('catalog/product')->load($item->getProductId());
+            $childProduct  = Mage::getModel('catalog/product_type_configurable')->getProductByAttributes(
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: You may need to do it for other products as well, our endpoints handle all products.

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php [hunk 2]
```diff
@@ -49,4 +50,21 @@ public function execute(Varien_Event_Observer $observer)
         $output = $observer->getData('output');
         $output->setData('items', $items);
     }
+
+    /**
+     * @param Mage_Wishlist_Model_Item $item
+     */
+    private function appendChildId($item)
+    {
+        $buyRequestItem = new Varien_Object($item->getBuyRequest()->getData());
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: isn't the buyRequest a VarienObject if you are calling getData() on it?

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php [hunk 3]
```diff
@@ -41,6 +41,7 @@ public function execute(Varien_Event_Observer $observer)
         $items = array();
         /** @var Mage_Wishlist_Model_Item $item */
         foreach ($collection as $item) {
+            $this->appendChildId($item);
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: you don't want to do magic work here by reference, it's not quite clear when reading what is mapped

**Comment by augsteyer** [comment]
Don't forget to fix the pull request template that you keep ignoring :(

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php (hunk 2 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: ^^ - you are right

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php (hunk 1 above)
Comment (awesselburg) [reply src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: do you mean for example a group product?

there a re the child information in the buyRequest like:

"super_group": {
                "548": "8",
                "551": "3"
            },

I can map this ids to the child_id, but in this case i sould rename this key to child_ids and return a array of items.

Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: Not a bad idea. Can you check the bundled as well? Please note that your tests are failing as you need to update them too, presumably.

**Comment by augsteyer** [comment]
you may need to stylize the collection.json, too many changes (I cannot see what u had changed)

**Comment by awesselburg** [comment]
> you may need to stylize the collection.json, too many changes (I cannot see what u had changed)

Please update your Postman App, there is a new Version available. I guess this is the reason for so many changes.

**Comment by augsteyer** [comment]
it's a bit long to explain, just leave the PR as is, I will finish it up tomorrow.

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php [hunk 4]
```diff
@@ -43,10 +43,47 @@ public function execute(Varien_Event_Observer $observer)
...
+     * Returns the product child id based on parents super attributes if exist
+     *
+     * @param Mage_Wishlist_Model_Item $item
+     *
+     * @return null | int
+     */
+    private function getChildId($item)
+    {
+        $superAttribute = $item->getBuyRequest()->getSuperAttribute();
+        if (is_array($superAttribute) && count($superAttribute) && count(array_values($superAttribute))) {
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php]: any particular scenario that the other 2 statements don't cover? referring to `count(array_values($superAttribute))`

**Pull request merged**